rasa/core/tracker_store.py

```diff
@@ -219,7 +219,10 @@ def retrieve_full_tracker(
         return self.retrieve(conversation_id)
 
     def stream_events(self, tracker: DialogueStateTracker) -> None:
-        """Streams events to a message broker"""
+        """Streams events to a message broker."""
+        if self.event_broker is None:
```

Comment:
You're right, it's not possible, because `stream_events` is being used only in the implementations of `save` method and this already checks that `self.event_broker is not None`, even when it's None, the `save` concretion continues running. I would say we don't need to raise an Exception, but we could possible remove all the `if` checks from `save` method, keep this check in `stream_events` and just run `stream_events` in `save` method?

Reply:
All comments have been addressed apart from this one, please let me know your thoughts here?

Reply:
so I think that we can keep this new condition here and cleanup the ones surrounding the calls to `stream_events`?